hal/api/EventAdapter.h

```diff
@@ -0,0 +1,316 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_EVENT_ADAPTER_H
+#define MBED_EVENT_ADAPTER_H
+
+#include "Callback.h"
+#if MBED_CONF_MBED_EVENTS_PRESENT
```

Comment:
The EventAdapter class is acting as a replacement for Callback, which means it's used by the hal even in the mbed 2 release. 

The MBED_CONF_MBED_EVENTS_PRESENT is used for EventAdapter similarly to the MBED_CONF_MBED_RTOS_PRESENT for mutexes.